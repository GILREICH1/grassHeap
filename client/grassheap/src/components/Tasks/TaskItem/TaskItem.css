.taskItem_crop_icon {
  display: inline-block;
  width: 1rem;

}
