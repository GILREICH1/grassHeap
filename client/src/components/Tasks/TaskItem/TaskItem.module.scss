.taskItem_crop_icon {
  display: inline-block;
  width: 1rem;
}

:disabled {
  cursor: not-allowed;
}

.tooltiptext {
  visibility: hidden;
}
.disabled:hover .tooltiptext {
  visibility: visible;
  z-index: 1;
}

.TaskItem {
  h3::first-letter {
    text-transform: capitalize;
  }
}
