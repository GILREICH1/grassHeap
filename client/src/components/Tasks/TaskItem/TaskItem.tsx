/* eslint-disable no-unused-vars */
import React from 'react';
import { Task } from '../../../common/types';
import styles from './TaskItem.module.scss';

interface TaskItemProps {
  task: Task;
  deleteThisTask: (_id: string) => void;
}

function TaskItem({ task, deleteThisTask }: TaskItemProps): JSX.Element {
  const taskIcon = `https://www.growstuff.org/crops/${task.crop}.svg`;
  const backUpIcon =
    'https://www.growstuff.org/assets/icons/planting-ce51a46e4a6edd740221f4a98f2e630a944e30ca040b9000d25179c8f5bc17e8.svg';

  return (
    <div className={styles.TaskItem}>
      <h3>
        {task.crop}{' '}
        <img
          className={styles.taskItem_crop_icon}
          src={taskIcon}
          onError={e => ((e.target as HTMLImageElement).src = backUpIcon)}
        />
      </h3>
      <p>{task.task}</p>
      {task.userCreated ? (
        <button onClick={() => deleteThisTask(task._id!)}>X</button>
      ) : null}
    </div>
  );
}

export default TaskItem;
