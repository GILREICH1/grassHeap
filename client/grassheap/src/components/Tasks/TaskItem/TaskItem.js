import React from "react";
import "./TaskItem.css";

function TaskItem({ task, deleteThisTask }) {
  const taskIcon = `https://www.growstuff.org/crops/${task.crop}.svg`;
  const backUpIcon = `https://www.growstuff.org/assets/icons/planting-ce51a46e4a6edd740221f4a98f2e630a944e30ca040b9000d25179c8f5bc17e8.svg`;
  const disabledButton = (
    <div>
      <button className="disabled" disabled>
        X
      </button>
      <span className="tooltiptext">cannot delete default tasks</span>
    </div>
  );

  return (
    <div>
      <h3>
        {task.crop}{" "}
        <img
          className="taskItem_crop_icon"
          src={taskIcon}
          onError={(e) => (e.target.src = backUpIcon)}
        ></img>
      </h3>
      <p>{task.task}</p>
      {task.userCreated ? (
        <button onClick={() => deleteThisTask(task._id)}>X</button>
      ) : (
        disabledButton
      )}
    </div>
  );
}

export default TaskItem;
