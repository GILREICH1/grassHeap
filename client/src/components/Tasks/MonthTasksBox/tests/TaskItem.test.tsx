import React from 'react';
import renderer from 'react-test-renderer';
import { render, unmountComponentAtNode } from 'react-dom';
import { act } from 'react-dom/test-utils';
import mocks from './mocks';
import TaskItem from '../../TaskItem/TaskItem';

describe('MonthTasksBox tests', () => {
  let container: Element;
  const mockedProps = {
    deleteThisTask: jest.fn(),
  };

  beforeEach(() => {
    // setup a DOM element as a render target
    container = document.createElement('div');
    document.body.appendChild(container);
  });

  afterEach(() => {
    // cleanup on exiting
    if (container) {
      unmountComponentAtNode(container);
      container.remove();
    }
  });

  describe('layout', () => {
    test('snapshot', () => {
      const tree = renderer
        .create(
          <TaskItem
            deleteThisTask={mockedProps.deleteThisTask}
            task={mocks.taskList[0]}
          />,
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe('props', () => {
    test('backend task display', async () => {
      act(() => {
        render(
          <TaskItem
            deleteThisTask={mockedProps.deleteThisTask}
            task={mocks.taskList[1]}
          />,
          container,
        );
      });

      expect(container.textContent).toContain(mocks.taskList[1]['task']);
      expect(container.textContent).toContain(mocks.taskList[1]['crop']);

      const deleteBtn = container.querySelector('.div1 button');
      expect(deleteBtn).toBeFalsy();
    });

    test('user created task display', async () => {
      act(() => {
        render(
          <TaskItem
            deleteThisTask={mockedProps.deleteThisTask}
            task={mocks.taskList[0]}
          />,
          container,
        );
      });

      const pElement = container.querySelector('p');
      expect(pElement?.textContent).toBe(mocks.taskList[0]['task']);

      const deleteBtn = container.querySelector('.div1 button');
      expect(deleteBtn).toBeTruthy();
    });
  });
});
