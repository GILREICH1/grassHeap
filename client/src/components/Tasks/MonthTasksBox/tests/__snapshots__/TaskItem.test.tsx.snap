// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MonthTasksBox tests layout snapshot 1`] = `
<div
  className="TaskItem"
>
  <div
    className="div2"
  >
    <h3>
      squash
       
      <img
        className="taskItem_crop_icon"
        onError={[Function]}
        src="https://www.growstuff.org/crops/squash.svg"
      />
    </h3>
  </div>
  <div
    className="div3"
  >
    <p>
      test task
    </p>
  </div>
  <div
    className="div1"
  >
    <div
      className="bar"
    />
    <button
      onClick={[Function]}
    >
      X
    </button>
  </div>
</div>
`;
